test/unit/libraries/string-transform.spec.js

```
@@ -0,0 +1,69 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+// @flow
+
+import { describe, it, expect } from '../../helpers/dependencies'
+import { applyTransformation, filterByString, prependWithFn } from '../../../src/libraries/string-transform'
+import { CallbackRecorder } from '../../helpers/utils'
+
+describe('prependWithFn', () => {
+  it('prepends each time with fn execution result', () => {
+    let calledTimes = 0
+    const prependFn = () => {
+      calledTimes += 1
+      return calledTimes.toString()
+    }
+    const transform = prependWithFn(prependFn)
+    expect(transform('data')).to.eql('1data')
+    expect(transform('data')).to.eql('2data')
+  })
+})
+
+describe('filterByString', () => {
+  it('returns same string if filter matches', () => {
+    expect(filterByString('a')('bald')).to.eql('bald')
+  })
+  it('returns nothing if filter doesn\'t match', () => {
+    expect(filterByString('a')('bold')).to.be.undefined
+  })
+})
+
+describe('applyTransformation', () => {
+  let called
+  const cbRec1 = new CallbackRecorder()
+  const cbRec2 = new CallbackRecorder()
+  const transform = (data) => {
+    if (!called) {
+      called = true
+      return data + '+'
+    }
+  }
+  const transformReturnNull = () => {
+  }
+
+  it('runs transformation function and pipes its output to callback', () => {
+    applyTransformation(transform, cbRec1.getCallback())('data')
+    expect(cbRec1.invoked).to.be.true
+    expect(cbRec1.argument).to.eql('data+')
+  })
+
+  it('does not call callback if transformation returns undefined or null', () => {
+    applyTransformation(transformReturnNull, cbRec2.getCallback())('data')
+    expect(cbRec2.invoked).to.be.false
+  })
```

Is this expected behaviour that transformation can return `null` / `unexpected`? If not, we should throw an exception, because this can hide a bug.

---

`null || undefined` is an expected as a return type from transformation, in that case we don't call the `next()` callback